python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -726,14 +768,24 @@ def mult(df):
         if cudf.api.types.is_scalar(chunk_results[0]):
             result = cudf.Series(chunk_results, index=group_names)
             result.index.names = self.grouping.names
-        elif isinstance(chunk_results[0], cudf.Series):
-            if isinstance(self.obj, cudf.DataFrame):
+        else:
+            if isinstance(chunk_results[0], cudf.Series) and isinstance(
+                self.obj, cudf.DataFrame
+            ):
                 result = cudf.concat(chunk_results, axis=1).T
                 result.index.names = self.grouping.names
             else:
                 result = cudf.concat(chunk_results)
-        else:
-            result = cudf.concat(chunk_results)
+                if self._group_keys:
+                    result_index = cudf.MultiIndex.from_frame(
```

Comment:
Is there any way to use an internal constructor  (e.g., `_from_data`) here?

Reply:
Yeah, we can. Let me make the change and get back

Reply:
Done, changed this to use `_from_data`